Tasks/AzureFileCopy/AzureUtilityGTE1.0.ps1

```diff
@@ -340,4 +340,108 @@ function Remove-AzureMachineCustomScriptExtension
+            $winRMConfigRule = Get-AzureRmNetworkSecurityRuleConfig -NetworkSecurityGroup $securityGroup -Name $ruleName -EA SilentlyContinue
+            Write-Verbose -Verbose "[Azure Call]Got network security rule config $ruleName under security group $securityGroupName"
+        }
+        catch
+        {
+            #Ignore the exception
+        }
+
+        # Add the network security rule if it doesn't exists
+        if(-not $winRMConfigRule)
+        {            
+            for($i=1; $i -le 3; $i++)
+            {
+                try
+                {
+                    Write-Verbose -Verbose "[Azure Call]Adding inbound network security rule config $ruleName with priority $rulePriotity for port $winrmHttpsPort under security group $securityGroupName"
+                    $result = Add-AzureRmNetworkSecurityRuleConfig -NetworkSecurityGroup $securityGroup -Name $ruleName -Direction Inbound -Access Allow -SourceAddressPrefix '*' -SourcePortRange '*' -DestinationAddressPrefix '*' -DestinationPortRange $winrmHttpsPort -Protocol * -Priority $rulePriotity | Set-AzureRmNetworkSecurityGroup
+                    Write-Verbose -Verbose "[Azure Call]Added inbound network security rule config $ruleName with priority $rulePriotity for port $winrmHttpsPort under security group $securityGroupName"                         
+                }
+                catch
+                {           
+                    $newPort = [convert]::ToInt32($rulePriotity, 10) + 50;
+                    $rulePriotity = $newPort.ToString()
+
+                    Write-Verbose -Verbose "[Azure Call]Getting network security group $securityGroupName in resource group $resourceGroupName"
+                    $securityGroup = Get-AzureRmNetworkSecurityGroup -ResourceGroupName $resourceGroupName -Name $securityGroupName
+                    Write-Verbose -Verbose "[Azure Call]Got network security group $securityGroupName in resource group $resourceGroupName"
+
+                    continue
+                }           
+                        
+                Write-Verbose -Verbose "Successfully added the network security group rule $ruleName with priority $rulePriotity for port $winrmHttpsPort"
+                break             
+            }
+        }
+        else
+        {
+            Write-Verbose -Verbose "Skipping the addition of network security rule '$ruleName' for $securityGroupName as it already exists"
+        }
+    }
+}
+
+function Get-NetworkSecurityGroup
+{
+     param([string] $resourceGroupName,
+           [string] $vmId)
+
+    $securityGroup = $null
+
+    if(-not [string]::IsNullOrEmpty($resourceGroupName) -and -not [string]::IsNullOrEmpty($vmId))
+    {
+        Write-Verbose -Verbose "[Azure Call]Getting network interfaces in resource group $resourceGroupName for vm $vmId"
+        $networkInterface = Get-AzureRmNetworkInterface -ResourceGroupName $resourceGroupName | Where-Object { $_.VirtualMachine.Id -eq $vmId }
```

Comment:
Can this be a list? I know you are not dealing with multiple nics. But if there is any VM returning multiple nic, you might want to gracefully exit.